streaming/src/streaming_message.h

```diff
@@ -0,0 +1,88 @@
+#ifndef RAY_STREAMING_MESSAGE_H
+#define RAY_STREAMING_MESSAGE_H
+
+#include <memory>
+
+#include "streaming_serializable.h"
+
+namespace ray {
+namespace streaming {
+
+class StreamingMessage;
+
+typedef std::shared_ptr<StreamingMessage> StreamingMessagePtr;
+
+enum class StreamingMessageType : uint32_t {
+  Barrier = 1,
+  Message = 2,
+  MIN = Barrier,
+  MAX = Message
+};
+
+constexpr uint32_t kMessageHeaderSize =
+    sizeof(uint32_t) + sizeof(uint64_t) + sizeof(StreamingMessageType);
+
+/*
+        +----------------+
+        | DataSize=U32   |
+        +----------------+
+        | MessageId=U64  |
+        +----------------+
+        | MessageType=U32|
+        +----------------+
+        | Data=var       |
+        +----------------+
```

Comment:
Better describe what a `StreamingMessage` is and  what it's used for, before giving this structure.